diff --git a/core/src/main/java/com/alibaba/fastjson2/JSON.java b/core/src/main/java/com/alibaba/fastjson2/JSON.java
index 1ae412775c..50f07c9ab0 100644
--- a/core/src/main/java/com/alibaba/fastjson2/JSON.java
+++ b/core/src/main/java/com/alibaba/fastjson2/JSON.java
@@ -108,7 +108,7 @@ public interface JSON {
     /**
      * fastjson2 version name
      */
-    String VERSION = "2.0.61";
+    String VERSION = "2.0.64";
 
     /**
      * Parses the json string as a {@link JSONArray} or {@link JSONObject}.
